src/db/db_backup.py

```
@@ -13,42 +11,55 @@
+
+    # 3. Execute the backup
+    try:
+        subprocess.run(command, env=env, check=True, capture_output=True, text=True)
+        logger.info(
+            "info_type=db_backup ; action=before_status_change ; database=%s ; backup=%s",
+            settings.db_name,
+            backup_path,
+        )
+    except subprocess.CalledProcessError as e:
+        logger.error("Database backup failed: %s", e.stderr)
+        raise RuntimeError(f"PostgreSQL backup failed: {e.stderr}")
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)